community/kernel/src/main/java/org/neo4j/kernel/api/security/AccessMode.java

```diff
@@ -228,142 +175,56 @@ public boolean allowsSchemaWrites()
-            public AuthorizationViolationException onViolation( String msg )
-            {
-                return new AuthorizationViolationException( msg );
-            }
-        },
-
-        /**
-         * Allows writing data, as well as reading data and schema.
-         * NOTE: This is a special mode that will override the original access mode when put as a temporary restriction
-         * on a transaction, potentially elevating the access mode to give write access to a transaction that did not
-         * have write access before.
-         */
-        OVERRIDE_WRITE
+        @Override
+        public AuthorizationViolationException onViolation( String msg )
         {
-            @Override
-            public boolean allowsReads()
-            {
-                return true;
-            }
-
-            @Override
-            public boolean allowsWrites()
-            {
-                return true;
-            }
-
-            @Override
-            public boolean allowsSchemaWrites()
-            {
-                return false;
-            }
-
-            @Override
-            public boolean overrideOriginalMode()
-            {
-                return true;
-            }
-
-            @Override
-            public AuthorizationViolationException onViolation( String msg )
-            {
-                return new AuthorizationViolationException( msg );
-            }
-        },
+            return new AuthorizationViolationException( msg );
+        }
 
-        /**
-         * Allows writing and reading data and schema.
-         * NOTE: This is a special mode that will override the original access mode when put as a temporary restriction
-         * on a transaction, potentially elevating the access mode to give schema access to a transaction that did not
-         * have schema access before.
-         */
-        OVERRIDE_SCHEMA
+        @Override
+        public AccessMode getSnapshot()
         {
-            @Override
-            public boolean allowsReads()
-            {
-                return true;
-            }
-
-            @Override
-            public boolean allowsWrites()
-            {
-                return true;
-            }
-
-            @Override
-            public boolean allowsSchemaWrites()
-            {
-                return true;
-            }
-
-            @Override
-            public boolean overrideOriginalMode()
-            {
-                return true;
-            }
-
-            @Override
-            public AuthorizationViolationException onViolation( String msg )
-            {
-                return new AuthorizationViolationException( msg );
-            }
-        },
-
+            return this;
         }
+    }
 
     boolean allowsReads();
     boolean allowsWrites();
     boolean allowsSchemaWrites();
-    boolean overrideOriginalMode();
+
     AuthorizationViolationException onViolation( String msg );
     String name();
+
+    default String username()
```

Comment:
We plan on separating the concepts of AuthSubject and AccessMode soon, so we rather make the docs effort then.